src/vm/interpreter.cpp

```diff
@@ -1431,8 +1431,8 @@ CorJitResult Interpreter::GenerateInterpreterStub(CEEInfo* comp,
         sl.X86EmitPopReg(kEBP);
         sl.X86EmitReturn(static_cast<WORD>(argState.callerArgStackSlots * sizeof(void*)));
 #elif defined(_AMD64_)
-        // EDX has "ilArgs" i.e., just the point where registers have been homed.
-        sl.X86EmitIndexLeaRSP(kEDX, static_cast<X86Reg>(kESP_Unsafe), 8);
+        // Pass "ilArgs", i.e. just the point where registers have been homed, as 2nd arg
+        sl.X86EmitIndexLeaRSP(kRSI, static_cast<X86Reg>(kESP_Unsafe), 8);
```

Comment:
No, it should be kRDX, I guess kEDX and kRDX just have the same value and so there was no problem.

The amd64 calling convention is very different between windows and Unix. Windows use just four general purpose registers - RCX, RDX, R8, R9 for arg passing, while Unix uses six - RDI, RSI, RDX, RCX, R8 and R9. Moreover, the rules for passing structs in registers and on stack are completely different between the platforms.